gc3libs/backends/batch.py

```diff
@@ -157,6 +165,44 @@ def __init__(self, name,
                 "Unknown transport '%s'" % transport)
         self.accounting_delay = accounting_delay
 
+
+    @property
+    def spooldir(self):
+        """
+        Root folder for all working directories of GC3Pie tasks.
+
+        When this backend executes a task, it first creates a temporary
+        subdirectory of this folder, then launches commands in there.
+
+        If not explicitly set (e.g. at construction time), the "spool
+        directory" will be given a default value according to the logic of
+        :meth:`_init_spooldir`:
+
+        * If the remote environment variable ``TMPDIR`` is set and points to an
+          existing directory, that value is used;
+        * otherwise, the hard-coded default ``/var/tmp`` is used instead.
+        """
+        if not self._spooldir:
+            self._init_spooldir()
+        return self._spooldir
+
+    @spooldir.setter
+    def spooldir(self, value):
+        self._spooldir = value
+
+    def _init_spooldir(self):
+        """Set `self.spooldir` to a sensible value."""
+        self.transport.connect()
+        rc, stdout, stderr = self.transport.execute_command(
+            'cd "{0}" && pwd'.format(gc3libs.Default.SPOOLDIR))
```

Comment:
...and since `$TMPDIR` is no longer used, running this command makes little sense.

Remove `self._init_spooldir()` altogether?